### PR TITLE
chore: Migrate transfer endpoints to v5 SDK

### PR DESCRIPTION
**Pull request by arcoraven**
Maps the following routes to the v5 SDK methods:

- erc20/transfer -> transfer
- erc20/transferFrom -> transferFrom
- erc721/transfer -> transferFrom
- erc721/transferFrom -> transferFrom
- erc1155/transfer -> safeTransferFrom
- erc1155/transferFrom -> safeTransferFrom

---

## PR-Codex overview
This PR focuses on enhancing the functionality and testing of ERC20, ERC721, and ERC1155 token transfers, along with improving schema definitions and utility functions. It introduces new test cases and refines existing code for better clarity and usability.

### Detailed summary
- Added `TEST_ACCOUNT_B` to `test/e2e/utils/wallets.ts`.
- Introduced `HexSchema` in `src/server/schemas/address.ts`.
- Updated example values in comments for `amount` in `sdk/src/services/BackendWalletService.ts`.
- Changed import for `Address` in `test/e2e/tests/setup.ts`.
- Removed console logs and improved assertions in `test/e2e/tests/write.test.ts`.
- Refined parameter descriptions in various services (ERC20, ERC721, ERC1155).
- Added `data` parameter to transfer functions in ERC1155.
- Introduced `getEngineBackendWalletB` function in `test/e2e/utils/engine.ts`.
- Updated contract interaction methods to use `thirdweb` client in multiple files.
- Enhanced transfer tests for ERC20, ERC721, and ERC1155 with new assertions and setup logic.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/server/schemas/number.ts [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
   examples: ["0.1"],
 });
 
 export const WeiAmountStringSchema = Type.RegExp(/^\d+$/, {
-  description: 'An amount in wei (no decimals). Example: "100000000"',
+  description: 'An amount in wei (no decimals). Example: "50000000000"',
   examples: ["50000000000"],
 });
+
+export const NumberStringSchema = Type.RegExp(/^\d+$/, {
```
Comment (arcoraven) [inline src/server/schemas/number.ts:13]: Useful for any generic non-decimal number input like tokenId or amount that must be a whole number (e.g. transferring NFTs).

File: src/server/routes/contract/extensions/erc1155/write/transfer.ts [hunk 2]
```diff
@@ -67,24 +76,38 @@ export async function erc1155transfer(fastify: FastifyInstance) {
...
+        to: getChecksumAddress(to),
+        tokenId: BigInt(tokenId),
+        value: BigInt(amount),
+        data: "0x",
+      });
+
+      const queueId = await queueTransaction({
+        transaction,
+        fromAddress: getChecksumAddress(walletAddress),
+        toAddress: getChecksumAddress(contractAddress),
```
Comment (arcoraven) [inline src/server/routes/contract/extensions/erc1155/write/transfer.ts:102]: Making sure the from/to here is always backend wallet -> contract (as opposed to above when building the transaction where from/to are the sender/recipient addresses).

File: src/server/routes/contract/extensions/erc1155/write/transfer.ts [hunk 3]
```diff
@@ -67,24 +76,38 @@ export async function erc1155transfer(fastify: FastifyInstance) {
...
-        simulateTx,
-        extension: "erc1155",
-        idempotencyKey,
+      const transaction = safeTransferFrom({
+        contract,
+        from: getChecksumAddress(walletAddress),
+        to: getChecksumAddress(to),
+        tokenId: BigInt(tokenId),
+        value: BigInt(amount),
+        data: "0x",
```
Comment (d4mr) [inline src/server/routes/contract/extensions/erc1155/write/transfer.ts:96]: I'd expose `data` as an optional body param

File: src/server/routes/contract/extensions/erc1155/write/transferFrom.ts [hunk 4]
```diff
@@ -74,29 +84,38 @@ export async function erc1155transferFrom(fastify: FastifyInstance) {
...
-        simulateTx,
-        extension: "erc1155",
-        idempotencyKey,
+      const transaction = safeTransferFrom({
+        contract,
+        from: getChecksumAddress(from),
+        to: getChecksumAddress(to),
+        tokenId: BigInt(tokenId),
+        value: BigInt(amount),
+        data: "0x",
```
Comment (d4mr) [inline src/server/routes/contract/extensions/erc1155/write/transferFrom.ts:104]: expose data here too, and all the other routes

File: src/server/routes/contract/extensions/erc1155/write/transfer.ts [hunk 5]
```diff
@@ -67,24 +76,38 @@ export async function erc1155transfer(fastify: FastifyInstance) {
...
       });
-      const tx = await contract.erc1155.transfer.prepare(to, tokenId, amount);
 
-      const queueId = await queueTx({
-        tx,
-        chainId,
-        simulateTx,
-        extension: "erc1155",
-        idempotencyKey,
+      const transaction = safeTransferFrom({
```
Comment (d4mr) [inline src/server/routes/contract/extensions/erc1155/write/transfer.ts:90]: safeTransferFrom can fail if the recipient is a contract that does not implement the onReceived hook. This is generally desirable behaviour, but could fail transfers to smart accounts that do not implement this method. Most smart accounts do implement this though, including ours I think, but would be best to confirm once.

**Review by d4mr — APPROVED** [review]

File: test/e2e/utils/engine.ts [hunk 6]
```diff
@@ -44,19 +44,17 @@ export const createChain = async (engine: Engine) => {
...
-  ) {
-    console.log("Creating backend wallet");
-    const newWallet = await engine.backendWallet.import({
-      privateKey: ANVIL_PKEY_A,
-    });
-    console.log("Created new wallet", newWallet.result.walletAddress);
-
-    return getAddress(newWallet.result.walletAddress);
-  }
+  const { result } = await engine.backendWallet.import({
```
Comment (arcoraven) [inline test/e2e/utils/engine.ts:47]: Importing the same wallet into Engine is fine (it will dedupe)

**Review by d4mr — APPROVED** [review]

**Pull request merged**